akka-javasdk/src/main/java/akka/javasdk/client/ComponentClient.java

```diff
@@ -58,4 +61,20 @@ public interface ComponentClient {
 
   /** Select {@link akka.javasdk.view.View} as a call target component. */
   ViewClient forView();
+
+  /**
+   * Select {@link Agent} as a call target component.
+   *
+   * @param sessionId - The agent may participate in a session, which is used for the agent's
+   *     conversational memory.
+   */
+  AgentClient forAgent(Optional<String> sessionId);
```

Comment:
I'm still confused about the sessionId, I would expect to see the agentId here, or a combination of agentId and sessionId. Can we discuss a business case for it? So that I could organize my thoughts around it?

Reply:
I see the confusion. Maybe we should not pass the session id in the `forAgent` method. 
It makes it to similar to how we get an handle on the other components. 

Instead, we could do the following:


```java
componentClient
  .forAgent()
  .withSessionId(...) // opt'ing-in to participate on a session
  .method(NewAskAkkaAgent::ask)
  .invoke(request.question);
```

and then, when not taking part of a session


```java
componentClient
  .forAgent()
  .method(NewAskAkkaAgent::ask)
  .invoke(request.question);
```

Reply:
agentId is an alias for `@ComponentId`, i.e. a static identifier for the agent, such as planner-agent, booking-agent, activities-agent, weather-agent

sessionId is the (uuid) for the session that the agent may participate in. E.g. planning a specific trip to Paris for a specific user. We intend to use the sessionId for the agent's conversational memory. Several agents may participate in the same session, and thereby share memory.

Reply:
ok... I understand it now. Ok, in that case, I think that we shouldn't expose too much the `agentId`, because people might confuse it with `workflowId` or `entityId`.

Reply:
And yes, making the component client more explicit about the sessionId, is also a good idea. We can iterate over it.

Reply:
sessionId is similar to workflowId and entityId, so I don't think it's confusing to have it in the `forAgent`.
I also want it to be an explicit choice to opt-out of sessionId, and it would be easy to miss its existence if we made it too optional `withSessionId`.

Reply:
It's not the same thing. You can instantiate two completely different agents with the same session id and they will share the session memory. This is not an agent unique identifier and is completely different to how it works with entities and workflows.

It's rather an instruction to participate into something.

> I also want it to be an explicit choice to opt-out of sessionId, and it would be easy to miss its existence if we made it too optional withSessionId.

I don't feel that this is so important. Getting an agent working requires lots of interactions. If one forgets to configure their agent to participate on an session, they will find it out pretty fast. 

It's not that we will be corrupting data if we forget to join a session and it's not something irreversible either.

Reply:
Yeah, I think it's confusing because I come with the mindset from entities and workflows. And I think users will make this same association as well. They might create different IDs for each the Agents they are interacting, for example. It feels too much as the ID of the instance we are creating through the component client.

Your last proposal sounds much better to me. 
Make it mandatory and thoughtful decision. And it make it clear that is a matter of participating into something, and not an ID to instantiate something. 

I guess they need to be paired with `forStandaloneAgent` (no session) or `forAgent().standalone()`. Or maybe we should say "sessionless`?

Reply:
Ok, I'll make the change, and make it mandatory. We shouldn't have no session option because it should be extremely rare with an agent that doesn't have a conversation or collaborate with other agents. If that is still needed it's just a matter of creating a new uuid for that call.